src/EditorFeatures/CSharpTest/CodeActions/IntroduceVariable/IntroduceVariableTests.cs

```diff
@@ -968,7 +968,7 @@ static void Main<T>(string[] args)
         Goo([|(T)2.ToString()|]);
     }
 }",
-count: 2);
+count: 3);
```

Comment:
Is there a way to avoid this? Both of the "Introduce local for all occurrences" fixers here result in the same changes being applied.

Reply:
That was a question I had too, the problem is that the deep analysis is done only when the code analysis is run, not listed to be added to the popup menu. 

I guess I could extract all the code necessary to determine if any of the replacements are l-values to before adding the code fixes to the active list, but that would make it more expensive whether or not you want to actually run the code fix (every time you select something in fact). Not to mention the fact that the same work will have to run again when the code fix is done, so we're not saving any time when applying the fix either. 

Personally I'm okay-ish with the 2 entries if the lvalue replacement is actually needed (I don't think it is) given that it's noticeably cheaper.

Reply:
i would prefer we do the up front check.  i'm not really seeing why that would be a problem per se.